plugins/tekton/src/components/pipeline-topology/PipelineVisualizationCard.test.tsx

```diff
@@ -16,8 +17,41 @@ jest.mock('@backstage/plugin-catalog-react', () => ({
   }),
 }));
 
-describe('PipelineVisualization', () => {
-  it('should render the pipeline run visualization when pipelineRun exists', async () => {
+jest.mock('@backstage/core-plugin-api', () => ({
+  ...jest.requireActual('@backstage/core-plugin-api'),
+  useRouteRef: () => () => '/xyz',
+}));
+
+describe('PipelineVisualizationCard', () => {
+  it('should render the latest pipeline run visualization when pipelineRun exists', async () => {
+    const mockContextData = {
+      watchResourcesData: {
+        pipelineruns: {
+          data: mockKubernetesPlrResponse.pipelineruns,
+        },
+        taskruns: {
+          data: mockKubernetesPlrResponse.taskruns,
+        },
+      },
+      loaded: true,
+      responseError: '',
+      selectedClusterErrors: [],
+      clusters: [],
+      setSelectedCluster: () => {},
+    };
+    const { queryByTestId } = render(
+      <TektonResourcesContext.Provider value={mockContextData}>
+        <PipelineVisualizationCard linkTekton={false} />
+      </TektonResourcesContext.Provider>,
+    );
+    expect(
+      screen.getByText(mockKubernetesPlrResponse.pipelineruns[1].metadata.name),
```

Comment:
we can use `getByText` from the render 

`const { getByText } = render( ... )`